esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2234,16 +2234,23 @@ void CFileSprayEx::getDropZoneInfoByIP(const char* ip, const char* destFileIn, S
     if (!computer.length())
         return;
 
-    Owned<IConstDropZoneInfo> dropZone = env->getDropZoneByComputer(computer.str());
-    if (!dropZone)
-        return;
-    dropZone->getDirectory(directory);
+    Owned<IConstDropZoneInfo> dropZone;
 
     StringBuffer destFile;
     if (isAbsolutePath(destFileIn))
+    {
         destFile.set(destFileIn);
+        dropZone.setown(env->getDropZoneByAddressPath(ip, destFile.str()));
+        if (!dropZone)
+            return;
+        dropZone->getDirectory(directory);
+    }
     else
     {
+        dropZone.setown(env->getDropZoneByComputer(computer.str()));
```

Comment:
The getDropZoneByComputer() function gives back the default/first (in environment.xml) drop zone on the given IP only. (It is a kind of legacy stuff)

If the intention of this function call is to determine there is a DZ on the give IP then it is fine. 
Otherwise to get a particular DZ on a given IP you can use:
 - getDropZoneByComputer(const char * computer, const char * dzname)
or
- Owned<IConstDropZoneInfoIterator> zoneIt= getDropZoneIteratorByComputer(const char * computer) 
then iterate trough the drop zones with:
- ForEach(*zoneIt) {...}